Compatibility with errorprone 2.19+ / StringCaseLocaleUsage

<details>
<summary>main</summary>

```java
> Task :grpc-xds:compileJava
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java:100: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
        ? altNameFromCert.toLowerCase().startsWith(sanToVerifyPrefix.toLowerCase())
                                     ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? altNameFromCert.toLowerCase(Locale.ROOT).startsWith(sanToVerifyPrefix.toLowerCase())' or '? altNameFromCert.toLowerCase(Locale.getDefault()).startsWith(sanToVerifyPrefix.toLowerCase())' or '? Ascii.toLowerCase(altNameFromCert).startsWith(sanToVerifyPrefix.toLowerCase())'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java:100: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
        ? altNameFromCert.toLowerCase().startsWith(sanToVerifyPrefix.toLowerCase())
                                                                                ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? altNameFromCert.toLowerCase().startsWith(sanToVerifyPrefix.toLowerCase(Locale.ROOT))' or '? altNameFromCert.toLowerCase().startsWith(sanToVerifyPrefix.toLowerCase(Locale.getDefault()))' or '? altNameFromCert.toLowerCase().startsWith(Ascii.toLowerCase(sanToVerifyPrefix))'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java:110: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? altNameFromCert.toLowerCase().endsWith(sanToVerifySuffix.toLowerCase())
                                         ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? altNameFromCert.toLowerCase(Locale.ROOT).endsWith(sanToVerifySuffix.toLowerCase())' or '? altNameFromCert.toLowerCase(Locale.getDefault()).endsWith(sanToVerifySuffix.toLowerCase())' or '? Ascii.toLowerCase(altNameFromCert).endsWith(sanToVerifySuffix.toLowerCase())'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java:110: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? altNameFromCert.toLowerCase().endsWith(sanToVerifySuffix.toLowerCase())
                                                                                  ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? altNameFromCert.toLowerCase().endsWith(sanToVerifySuffix.toLowerCase(Locale.ROOT))' or '? altNameFromCert.toLowerCase().endsWith(sanToVerifySuffix.toLowerCase(Locale.getDefault()))' or '? altNameFromCert.toLowerCase().endsWith(Ascii.toLowerCase(sanToVerifySuffix))'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java:120: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? altNameFromCert.toLowerCase().contains(sanToVerifySubstring.toLowerCase())
                                         ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? altNameFromCert.toLowerCase(Locale.ROOT).contains(sanToVerifySubstring.toLowerCase())' or '? altNameFromCert.toLowerCase(Locale.getDefault()).contains(sanToVerifySubstring.toLowerCase())' or '? Ascii.toLowerCase(altNameFromCert).contains(sanToVerifySubstring.toLowerCase())'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java:120: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? altNameFromCert.toLowerCase().contains(sanToVerifySubstring.toLowerCase())
                                                                                     ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? altNameFromCert.toLowerCase().contains(sanToVerifySubstring.toLowerCase(Locale.ROOT))' or '? altNameFromCert.toLowerCase().contains(sanToVerifySubstring.toLowerCase(Locale.getDefault()))' or '? altNameFromCert.toLowerCase().contains(Ascii.toLowerCase(sanToVerifySubstring))'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/Matchers.java:276: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? args.toLowerCase().startsWith(prefix().toLowerCase())
                              ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? args.toLowerCase(Locale.ROOT).startsWith(prefix().toLowerCase())' or '? args.toLowerCase(Locale.getDefault()).startsWith(prefix().toLowerCase())' or '? Ascii.toLowerCase(args).startsWith(prefix().toLowerCase())'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/Matchers.java:276: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? args.toLowerCase().startsWith(prefix().toLowerCase())
                                                                ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? args.toLowerCase().startsWith(prefix().toLowerCase(Locale.ROOT))' or '? args.toLowerCase().startsWith(prefix().toLowerCase(Locale.getDefault()))' or '? args.toLowerCase().startsWith(Ascii.toLowerCase(prefix()))'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/Matchers.java:280: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? args.toLowerCase().endsWith(suffix().toLowerCase())
                              ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? args.toLowerCase(Locale.ROOT).endsWith(suffix().toLowerCase())' or '? args.toLowerCase(Locale.getDefault()).endsWith(suffix().toLowerCase())' or '? Ascii.toLowerCase(args).endsWith(suffix().toLowerCase())'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/internal/Matchers.java:280: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            ? args.toLowerCase().endsWith(suffix().toLowerCase())
                                                              ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '? args.toLowerCase().endsWith(suffix().toLowerCase(Locale.ROOT))' or '? args.toLowerCase().endsWith(suffix().toLowerCase(Locale.getDefault()))' or '? args.toLowerCase().endsWith(Ascii.toLowerCase(suffix()))'?
/Users/sergiitk/Development/grpc-java/xds/src/main/java/io/grpc/xds/XdsClusterResource.java:245: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
      if ((edsServiceName == null) && clusterName.toLowerCase().startsWith("xdstp:")) {
                                                             ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'if ((edsServiceName == null) && clusterName.toLowerCase(Locale.ROOT).startsWith("xdstp:")) {' or 'if ((edsServiceName == null) && clusterName.toLowerCase(Locale.getDefault()).startsWith("xdstp:")) {' or 'if ((edsServiceName == null) && Ascii.toLowerCase(clusterName).startsWith("xdstp:")) {'?
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
11 warnings

> Task :grpc-authz:compileJava
/Users/sergiitk/Development/grpc-java/authz/src/main/java/io/grpc/authz/AuthorizationPolicyTranslator.java:80: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
        || UNSUPPORTED_HEADERS.contains(key.toLowerCase())) {
                                                       ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '|| UNSUPPORTED_HEADERS.contains(key.toLowerCase(Locale.ROOT))) {' or '|| UNSUPPORTED_HEADERS.contains(key.toLowerCase(Locale.getDefault()))) {' or '|| UNSUPPORTED_HEADERS.contains(Ascii.toLowerCase(key))) {'?
1 warning

> Task :grpc-benchmarks:compileJava
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ServerConfiguration.java:72: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
            "TLS unsupported with the " + config.transport.name().toLowerCase() + " transport");
                                                                             ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '"TLS unsupported with the " + config.transport.name().toLowerCase(Locale.ROOT) + " transport");' or '"TLS unsupported with the " + config.transport.name().toLowerCase(Locale.getDefault()) + " transport");' or '"TLS unsupported with the " + Ascii.toLowerCase(config.transport.name()) + " transport");'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ServerConfiguration.java:131: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
        builder.append(transport.name().toLowerCase());
                                                   ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'builder.append(transport.name().toLowerCase(Locale.ROOT));' or 'builder.append(transport.name().toLowerCase(Locale.getDefault()));' or 'builder.append(Ascii.toLowerCase(transport.name()));'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ServerConfiguration.java:161: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
    TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase()) {
                                                                                           ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase(Locale.ROOT)) {' or 'TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase(Locale.getDefault())) {' or 'TRANSPORT("STR", Transport.getDescriptionString(), Ascii.toLowerCase(DEFAULT.transport.name())) {'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ServerConfiguration.java:164: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
        config.transport = Transport.valueOf(value.toUpperCase());
                                                              ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'config.transport = Transport.valueOf(value.toUpperCase(Locale.ROOT));' or 'config.transport = Transport.valueOf(value.toUpperCase(Locale.getDefault()));' or 'config.transport = Transport.valueOf(Ascii.toUpperCase(value));'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ServerConfiguration.java:200: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
      return name().toLowerCase();
                               ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'return name().toLowerCase(Locale.ROOT);' or 'return name().toLowerCase(Locale.getDefault());' or 'return Ascii.toLowerCase(name());'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/Transport.java:67: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
      builder.append(transport.name().toLowerCase());
                                                 ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'builder.append(transport.name().toLowerCase(Locale.ROOT));' or 'builder.append(transport.name().toLowerCase(Locale.getDefault()));' or 'builder.append(Ascii.toLowerCase(transport.name()));'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java:105: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
              "Transport " + config.transport.name().toLowerCase() + " does not support TLS.");
                                                                ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean '"Transport " + config.transport.name().toLowerCase(Locale.ROOT) + " does not support TLS.");' or '"Transport " + config.transport.name().toLowerCase(Locale.getDefault()) + " does not support TLS.");' or '"Transport " + Ascii.toLowerCase(config.transport.name()) + " does not support TLS.");'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java:169: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
    TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase()) {
                                                                                           ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase(Locale.ROOT)) {' or 'TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase(Locale.getDefault())) {' or 'TRANSPORT("STR", Transport.getDescriptionString(), Ascii.toLowerCase(DEFAULT.transport.name())) {'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java:172: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
        config.transport = Transport.valueOf(value.toUpperCase());
                                                              ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'config.transport = Transport.valueOf(value.toUpperCase(Locale.ROOT));' or 'config.transport = Transport.valueOf(value.toUpperCase(Locale.getDefault()));' or 'config.transport = Transport.valueOf(Ascii.toUpperCase(value));'?
/Users/sergiitk/Development/grpc-java/benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java:239: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
      return name().toLowerCase();
                               ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'return name().toLowerCase(Locale.ROOT);' or 'return name().toLowerCase(Locale.getDefault());' or 'return Ascii.toLowerCase(name());'?
10 warnings

> Task :grpc-interop-testing:compileJava
/Users/sergiitk/Development/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/TestCases.java:82: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
    return TestCases.valueOf(s.toUpperCase());
                                          ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'return TestCases.valueOf(s.toUpperCase(Locale.ROOT));' or 'return TestCases.valueOf(s.toUpperCase(Locale.getDefault()));' or 'return TestCases.valueOf(Ascii.toUpperCase(s));'?
/Users/sergiitk/Development/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java:376: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
      String strTestcase = testCase.name().toLowerCase();
                                                      ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'String strTestcase = testCase.name().toLowerCase(Locale.ROOT);' or 'String strTestcase = testCase.name().toLowerCase(Locale.getDefault());' or 'String strTestcase = Ascii.toLowerCase(testCase.name());'?
/Users/sergiitk/Development/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java:695: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
      String strTestcase = testCase.name().toLowerCase();
                                                      ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'String strTestcase = testCase.name().toLowerCase(Locale.ROOT);' or 'String strTestcase = testCase.name().toLowerCase(Locale.getDefault());' or 'String strTestcase = Ascii.toLowerCase(testCase.name());'?
/Users/sergiitk/Development/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/Http2TestCases.java:52: warning: [StringCaseLocaleUsage] Specify a `Locale` when calling `String#to{Lower,Upper}Case`. (Note: there are multiple suggested fixes; the third may be most appropriate if you're dealing with ASCII Strings.)
      return Http2TestCases.valueOf(s.toUpperCase());
                                                 ^
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'return Http2TestCases.valueOf(s.toUpperCase(Locale.ROOT));' or 'return Http2TestCases.valueOf(s.toUpperCase(Locale.getDefault()));' or 'return Http2TestCases.valueOf(Ascii.toUpperCase(s));'?
4 warnings
```
</details>

--- a/authz/src/main/java/io/grpc/authz/AuthorizationPolicyTranslator.java
+++ b/authz/src/main/java/io/grpc/authz/AuthorizationPolicyTranslator.java
@@ -33,6 +33,7 @@ import java.io.IOException;
 import java.util.ArrayList;
 import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 
 /**
@@ -77,7 +78,7 @@ class AuthorizationPolicyTranslator {
     }
     if (key.charAt(0) == ':'
         || key.startsWith("grpc-")
-        || UNSUPPORTED_HEADERS.contains(key.toLowerCase())) {
+        || UNSUPPORTED_HEADERS.contains(key.toLowerCase(Locale.ROOT))) {
       throw new IllegalArgumentException(String.format("Unsupported \"key\" %s", key));
     }
     List<String> valuesList = JsonUtil.getListOfStrings(header, "values");

--- a/benchmarks/src/main/java/io/grpc/benchmarks/Transport.java
+++ b/benchmarks/src/main/java/io/grpc/benchmarks/Transport.java
@@ -16,6 +16,8 @@
 
 package io.grpc.benchmarks;
 
+import java.util.Locale;
+
 /**
  * All of the supported transports.
  */
@@ -64,11 +66,16 @@ public enum Transport {
       if (!first) {
         builder.append("\n");
       }
-      builder.append(transport.name().toLowerCase());
+      builder.append(transport);
       builder.append(": ");
       builder.append(transport.description);
       first = false;
     }
     return builder.toString();
+  }
+
+  @Override
+  public String toString() {
+    return name().toLowerCase(Locale.ROOT);
   }
 }

--- a/benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java
+++ b/benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java
@@ -31,6 +31,7 @@ import java.io.IOException;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.LinkedHashSet;
+import java.util.Locale;
 import java.util.Set;
 
 /**
@@ -102,7 +103,7 @@ public class ClientConfiguration implements Configuration {
       if (config.tls) {
         if (!config.transport.tlsSupported) {
           throw new IllegalArgumentException(
-              "Transport " + config.transport.name().toLowerCase() + " does not support TLS.");
+              "Transport " + config.transport + " does not support TLS.");
         }
       }
 
@@ -166,10 +167,10 @@ public class ClientConfiguration implements Configuration {
         config.testca = parseBoolean(value);
       }
     },
-    TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase()) {
+    TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.toString()) {
       @Override
       protected void setClientValue(ClientConfiguration config, String value) {
-        config.transport = Transport.valueOf(value.toUpperCase());
+        config.transport = Transport.valueOf(value.toUpperCase(Locale.ROOT));
       }
     },
     DURATION("SECONDS", "Duration of the benchmark.", "" + DEFAULT.duration) {
@@ -236,7 +237,7 @@ public class ClientConfiguration implements Configuration {
 
     @Override
     public String getName() {
-      return name().toLowerCase();
+      return name().toLowerCase(Locale.ROOT);
     }
 
     @Override

--- a/benchmarks/src/main/java/io/grpc/benchmarks/qps/ServerConfiguration.java
+++ b/benchmarks/src/main/java/io/grpc/benchmarks/qps/ServerConfiguration.java
@@ -29,6 +29,7 @@ import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
 
 /**
  * Configuration options for benchmark servers.
@@ -69,7 +70,7 @@ class ServerConfiguration implements Configuration {
     protected ServerConfiguration build0(ServerConfiguration config) {
       if (config.tls && !config.transport.tlsSupported) {
         throw new IllegalArgumentException(
-            "TLS unsupported with the " + config.transport.name().toLowerCase() + " transport");
+            "TLS unsupported with the " + config.transport + " transport");
       }
 
       // Verify that the address type is correct for the transport type.
@@ -109,6 +110,11 @@ class ServerConfiguration implements Configuration {
       this.socketAddressValidator = socketAddressValidator;
     }
 
+    @Override
+    public String toString() {
+      return name().toLowerCase(Locale.ROOT);
+    }
+
     /**
      * Validates the given address for this transport.
      *
@@ -128,7 +134,7 @@ class ServerConfiguration implements Configuration {
         if (!first) {
           builder.append("\n");
         }
-        builder.append(transport.name().toLowerCase());
+        builder.append(transport);
         builder.append(": ");
         builder.append(transport.description);
         first = false;
@@ -158,10 +164,10 @@ class ServerConfiguration implements Configuration {
         config.tls = parseBoolean(value);
       }
     },
-    TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.name().toLowerCase()) {
+    TRANSPORT("STR", Transport.getDescriptionString(), DEFAULT.transport.toString()) {
       @Override
       protected void setServerValue(ServerConfiguration config, String value) {
-        config.transport = Transport.valueOf(value.toUpperCase());
+        config.transport = Transport.valueOf(value.toUpperCase(Locale.ROOT));
       }
     },
     DIRECTEXECUTOR("", "Don't use a threadpool for RPC calls, instead execute calls directly "
@@ -197,7 +203,7 @@ class ServerConfiguration implements Configuration {
 
     @Override
     public String getName() {
-      return name().toLowerCase();
+      return name().toLowerCase(Locale.ROOT);
     }
 
     @Override

--- a/build.gradle
+++ b/build.gradle
@@ -253,11 +253,6 @@ subprojects {
             options.errorprone.check("JavaUtilDate", CheckSeverity.OFF)
             // The warning fails to provide a source location
             options.errorprone.check("MissingSummary", CheckSeverity.OFF)
-
-            // TODO(https://github.com/grpc/grpc-java/issues/10372): remove when fixed.
-            if (JavaVersion.current().isJava11Compatible()) {
-              options.errorprone.check("StringCaseLocaleUsage", CheckSeverity.OFF)
-            }
         }
         tasks.named("compileTestJava").configure {
             // LinkedList doesn't hurt much in tests and has lots of usages

--- a/interop-testing/src/main/java/io/grpc/testing/integration/Http2TestCases.java
+++ b/interop-testing/src/main/java/io/grpc/testing/integration/Http2TestCases.java
@@ -17,6 +17,7 @@
 package io.grpc.testing.integration;
 
 import com.google.common.base.Preconditions;
+import java.util.Locale;
 
 /**
  * Enum of HTTP/2 interop test cases.
@@ -49,7 +50,7 @@ public enum Http2TestCases {
   public static Http2TestCases fromString(String s) {
     Preconditions.checkNotNull(s, "s");
     try {
-      return Http2TestCases.valueOf(s.toUpperCase());
+      return Http2TestCases.valueOf(s.toUpperCase(Locale.ROOT));
     } catch (IllegalArgumentException ex) {
       throw new IllegalArgumentException("Invalid test case: " + s);
     }

--- a/interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java
+++ b/interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java
@@ -434,7 +434,7 @@ public class StressTestClient {
   private static String validTestCasesHelpText() {
     StringBuilder builder = new StringBuilder();
     for (TestCases testCase : TestCases.values()) {
-      String strTestcase = testCase.name().toLowerCase();
+      String strTestcase = testCase.toString();
       builder.append("\n      ")
           .append(strTestcase)
           .append(": ")

--- a/interop-testing/src/main/java/io/grpc/testing/integration/TestCases.java
+++ b/interop-testing/src/main/java/io/grpc/testing/integration/TestCases.java
@@ -17,6 +17,7 @@
 package io.grpc.testing.integration;
 
 import com.google.common.base.Preconditions;
+import java.util.Locale;
 
 /**
  * Enum of interop test cases.
@@ -79,6 +80,11 @@ public enum TestCases {
    */
   public static TestCases fromString(String s) {
     Preconditions.checkNotNull(s, "s");
-    return TestCases.valueOf(s.toUpperCase());
+    return TestCases.valueOf(s.toUpperCase(Locale.ROOT));
+  }
+
+  @Override
+  public String toString() {
+    return name().toLowerCase(Locale.ROOT);
   }
 }

--- a/interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java
+++ b/interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java
@@ -697,7 +697,7 @@ public class TestServiceClient {
   private static String validTestCasesHelpText() {
     StringBuilder builder = new StringBuilder();
     for (TestCases testCase : TestCases.values()) {
-      String strTestcase = testCase.name().toLowerCase();
+      String strTestcase = testCase.toString();
       builder.append("\n      ")
           .append(strTestcase)
           .append(": ")

--- a/xds/src/main/java/io/grpc/xds/XdsClusterResource.java
+++ b/xds/src/main/java/io/grpc/xds/XdsClusterResource.java
@@ -257,7 +257,7 @@ class XdsClusterResource extends XdsResourceType<CdsUpdate> {
         edsServiceName = edsClusterConfig.getServiceName();
       }
       // edsServiceName is required if the CDS resource has an xdstp name.
-      if ((edsServiceName == null) && clusterName.toLowerCase().startsWith("xdstp:")) {
+      if ((edsServiceName == null) && clusterName.toLowerCase(Locale.ROOT).startsWith("xdstp:")) {
         return StructOrError.fromError(
             "EDS service_name must be set when Cluster resource has an xdstp name");
       }

--- a/xds/src/main/java/io/grpc/xds/internal/Matchers.java
+++ b/xds/src/main/java/io/grpc/xds/internal/Matchers.java
@@ -22,6 +22,7 @@ import com.google.auto.value.AutoValue;
 import com.google.re2j.Pattern;
 import java.math.BigInteger;
 import java.net.InetAddress;
+import java.util.Locale;
 import javax.annotation.Nullable;
 
 /**
@@ -273,11 +274,11 @@ public final class Matchers {
             : exact().equals(args);
       } else if (prefix() != null) {
         return ignoreCase()
-            ? args.toLowerCase().startsWith(prefix().toLowerCase())
+            ? args.toLowerCase(Locale.ROOT).startsWith(prefix().toLowerCase(Locale.ROOT))
             : args.startsWith(prefix());
       } else if (suffix() != null) {
         return ignoreCase()
-            ? args.toLowerCase().endsWith(suffix().toLowerCase())
+            ? args.toLowerCase(Locale.ROOT).endsWith(suffix().toLowerCase(Locale.ROOT))
             : args.endsWith(suffix());
       } else if (contains() != null) {
         return args.contains(contains());

--- a/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java
+++ b/xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java
@@ -30,6 +30,7 @@ import java.security.cert.CertificateParsingException;
 import java.security.cert.X509Certificate;
 import java.util.Collection;
 import java.util.List;
+import java.util.Locale;
 import javax.annotation.Nullable;
 import javax.net.ssl.SSLEngine;
 import javax.net.ssl.SSLParameters;
@@ -97,7 +98,8 @@ final class XdsX509TrustManager extends X509ExtendedTrustManager implements X509
       return false;
     }
     return ignoreCase
-        ? altNameFromCert.toLowerCase().startsWith(sanToVerifyPrefix.toLowerCase())
+        ? altNameFromCert.toLowerCase(Locale.ROOT).startsWith(
+            sanToVerifyPrefix.toLowerCase(Locale.ROOT))
         : altNameFromCert.startsWith(sanToVerifyPrefix);
   }
 
@@ -107,7 +109,8 @@ final class XdsX509TrustManager extends X509ExtendedTrustManager implements X509
       return false;
     }
     return ignoreCase
-            ? altNameFromCert.toLowerCase().endsWith(sanToVerifySuffix.toLowerCase())
+            ? altNameFromCert.toLowerCase(Locale.ROOT).endsWith(
+                sanToVerifySuffix.toLowerCase(Locale.ROOT))
             : altNameFromCert.endsWith(sanToVerifySuffix);
   }
 
@@ -117,7 +120,8 @@ final class XdsX509TrustManager extends X509ExtendedTrustManager implements X509
       return false;
     }
     return ignoreCase
-            ? altNameFromCert.toLowerCase().contains(sanToVerifySubstring.toLowerCase())
+            ? altNameFromCert.toLowerCase(Locale.ROOT).contains(
+                sanToVerifySubstring.toLowerCase(Locale.ROOT))
             : altNameFromCert.contains(sanToVerifySubstring);
   }
 

